aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -744,15 +738,21 @@ TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tenso
       " and output2 has ", out2.sizes());
   TORCH_CHECK(out1.strides() == out2.strides(), "reduce_op(): expected both outputs to have same strides, but output1 has ", out1.strides(),
       " and output2 has ", out2.strides());
-  auto iter = TensorIterator();
+  auto iter = TensorIteratorConfig();
   iter.add_output(out1);
   iter.add_output(out2);
   iter.add_input(a);
-  iter.config_resize_outputs_ = false;
-  iter.config_is_reduction_ = true;
+  iter.dont_resize_outputs();
```

Comment:
Aside: I didn't update "dont_resize_outputs" but for consistency it's probably better as "resize_outputs(bool)," or you can make the other config flags more consistent with it: "disable_resizing_outputs(), enable_promoting_inputs()..."

Reply:
As preexisting conditions, I'm going to fix these in separate PRs.